src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/VsReferenceItem.cs

```diff
@@ -7,22 +7,20 @@
 namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore;
 
 /// <summary>
-///     Wraps a <see cref="ReferenceItem"/> instance to implement the <see cref="IVsReferenceItem"/>
+///     Wraps a <see cref="ReferenceItem"/> instance to implement the <see cref="IVsReferenceItem2"/>
 ///     interface for NuGet.
 /// </summary>
 [DebuggerDisplay("Name = {Name}")]
-internal class VsReferenceItem : IVsReferenceItem
+internal class VsReferenceItem : IVsReferenceItem2
 {
     private readonly ReferenceItem _referenceItem;
 
-    private VsReferenceProperties? _properties;
-
     public VsReferenceItem(ReferenceItem referenceItem)
     {
         _referenceItem = referenceItem;
     }
 
     public string Name => _referenceItem.Name;
 
-    public IVsReferenceProperties Properties => _properties ??= new VsReferenceProperties(_referenceItem.Properties);
+    public IReadOnlyDictionary<string, string?>? Properties => _referenceItem.Properties!;
```

Comment:
```suggestion
    public IReadOnlyDictionary<string, string?> Properties => _referenceItem.Properties;
```
`ReferenceItem.Properties` is non-null, right?

Reply:
According to your current implementation, yes. But I made it nullable in case project-system wants to avoid unnecessary allocations in the future when an item doesn't have any metadata defined. This will be the expected case for `<NuGetAuditSuppress` items, for example,

Reply:
> in case project-system wants to avoid unnecessary allocations in the future

We can always return a singleton empty object, like `Array.Empty<>()`.

Reply:
VS doesn't guarantee that only dotnet/project-system will call this API. If a 3rd party project system calls the API, NuGet can protect itself from nulls by expecting that a null might be possible.